packages/blade/src/components/SideNav/SideNav.tsx

```diff
@@ -0,0 +1,233 @@
+import type { SideNavContextType, SideNavProps } from './types';
+import {
+  classes,
+  COLLAPSED_L1_WIDTH,
+  EXPANDED_L1_WIDTH,
+  SKIP_NAV_ID,
+  useSideNavTransition,
+} from './tokens';
+import BaseBox from '~components/Box/BaseBox';
+import { makeSize, makeSpace } from '~utils';
+import { Drawer, DrawerBody, DrawerHeader } from '~components/Drawer';
+import { SkipNavContent, SkipNavLink } from '~components/SkipNav/SkipNav';
+import { useIsMobile } from '~utils/useIsMobile';
+import { getComponentId } from '~utils/isValidAllowedChildren';
+
+const {
+  COLLAPSED,
+  SHOW_WHEN_COLLAPSED,
+  HIDE_WHEN_COLLAPSED,
+  TRANSITIONING,
+  L1_ITEM_WRAPPER,
+} = classes;
+
+const StyledL1Menu = styled(BaseBox)((props) => {
+  const { l1Collapse, l1Expand } = useSideNavTransition();
+
+  return {
+    width: '100%',
+    transition: l1Expand,
+    [`& > .${L1_ITEM_WRAPPER}`]: {
+      padding: makeSpace(props.theme.spacing[4]),
+    },
+    [`.${SHOW_WHEN_COLLAPSED}`]: {
+      display: 'none',
+    },
+    [`&.${COLLAPSED}`]: {
+      width: makeSize(COLLAPSED_L1_WIDTH),
+      transition: l1Collapse,
+      [`& > .${L1_ITEM_WRAPPER}`]: {
+        padding: `${makeSpace(props.theme.spacing[4])} ${makeSpace(props.theme.spacing[3])}`,
+      },
+      [`&:not(.${TRANSITIONING}) .${HIDE_WHEN_COLLAPSED}`]: {
+        display: 'none',
+      },
+      [`&:not(.${TRANSITIONING}) .${SHOW_WHEN_COLLAPSED}`]: {
+        display: 'initial',
+      },
+    },
+  };
+});
+
+const getL1MenuClassName = ({
+  isL1Collapsed,
+  isL1Hovered,
+  isTransitioning,
+}: {
+  isL1Collapsed: boolean;
+  isL1Hovered: boolean;
+  isTransitioning: boolean;
+}): string => {
+  const isMenuCollapsed = isL1Collapsed && !isL1Hovered;
+
+  if (isMenuCollapsed) {
+    if (isTransitioning) {
+      return `${COLLAPSED} ${TRANSITIONING}`;
+    }
+
+    return COLLAPSED;
+  }
+
+  return '';
+};
+
+const SideNav = ({ children, isOpen, onDismiss }: SideNavProps): React.ReactElement => {
+  const l2PortalContainerRef = React.useRef(null);
+  const l1ContainerRef = React.useRef<HTMLDivElement>(null);
+  const [isL1Collapsed, setIsL1Collapsed] = React.useState(false);
+  const [isMobileL2Open, setIsMobileL2Open] = React.useState(false);
+  const [isL1Hovered, setIsL1Hovered] = React.useState(false);
+  const [isHoverAgainEnabled, setIsHoverAgainEnabled] = React.useState(true);
+  const [isTransitioning, setIsTransitioning] = React.useState(false);
+  const [l2DrawerTitle, setL2DrawerTitle] = React.useState('');
+
+  console.count('SideNav');
+  const isMobile = useIsMobile();
+
+  const closeMobileNav = (): void => {
+    if (isMobile) {
+      setIsMobileL2Open(false);
+      onDismiss?.();
+    }
+  };
+
+  const footerChild: React.ReactNode[] = [];
+  const restChild: React.ReactNode[] = [];
+
+  React.Children.map(children, (child) => {
+    if (getComponentId(child) === 'SideNavFooter') {
+      footerChild.push(child);
+    } else {
+      restChild.push(child);
+    }
+  });
```

Comment:
Instead of doing this can we add a wrapper in the public API of SideNav, eg `<SideNavBody>`

Reply:
done